src/charm.py

```diff
@@ -2,202 +2,124 @@
 
 import logging
 from string import Template
-
+from charms.observability_libs.v1.kubernetes_service_patch import KubernetesServicePatch
 from ops.charm import CharmBase
 from ops.main import main
-from ops.model import ActiveStatus, MaintenanceStatus
+from ops.model import ActiveStatus, WaitingStatus, ModelError
+from ops.pebble import ServiceStatus
+from lightkube.models.core_v1 import ServicePort
 
-from oci_image import OCIImageResource, OCIImageResourceError
+logger = logging.getLogger(__name__)
 
 
 class CoreDNSCharm(CharmBase):
+    """CoreDNS Sidecar Charm"""
+
+    _COREDNS_CONTAINER = "coredns"
+
     def __init__(self, *args):
         super().__init__(*args)
+
         if not self.unit.is_leader():
             # We can't do anything useful when not the leader, so do nothing.
-            self.model.unit.status = ActiveStatus()
+            self.model.unit.status = WaitingStatus("Waiting for leadership")
```

Comment:
if there are multiple coredns applications in the model -- the rest just say "Waiting for leadership"?  I mean?  I guess?

Reply:
I don't really think they should be marked as active if they are just sitting around waiting, but maybe waiting for leadership isnt the best choice of words for the status message. I think I stole that phrase from some other charm I saw

Reply:
The [Istio-gateway](https://github.com/canonical/istio-operators/blob/a9a65be14b8820429241b0d07879564742549070/charms/istio-gateway/src/charm.py#L20) and [istio-pilot](https://github.com/canonical/istio-operators/blob/a9a65be14b8820429241b0d07879564742549070/charms/istio-pilot/src/charm.py#L24) charms both use this status pattern.